main.rb

```diff
@@ -1,34 +1,112 @@
-require_relative 'class/person'
-require_relative 'class/student'
-require_relative 'class/teacher'
-require_relative 'class/capitalize_decorator'
-require_relative 'class/trimmer_decorator'
-require_relative 'class/classroom'
-
-# Create a Person object
-person = Person.new(22, name: 'maximilianus')
-puts "Person Name: #{person.correct_name}"
-
-capitalized_person = CapitalizeDecorator.new(person)
-puts "Capitalized Person Name: #{capitalized_person.correct_name}"
-
-capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
-puts "Capitalized and Trimmed Person Name: #{capitalized_trimmed_person.correct_name}"
-
-# Create a Classroom object
-classroom = Classroom.new('Classroom A')
-
-# Create a Student object and add it to the classroom
-student = Student.new(16, classroom)
-puts "Student ID: #{student.id}"
-puts "Student Name: #{student.name}"
-puts "Student Age: #{student.age}"
-puts "Can Use Services? #{student.can_use_services?}"
-puts student.play_hooky
-
-# Create a Teacher object
-teacher = Teacher.new(40, 'Math')
-puts "Teacher ID: #{teacher.id}"
-puts "Teacher Name: #{teacher.name}"
-puts "Teacher Age: #{teacher.age}"
-puts "Can Use Services? #{teacher.can_use_services?}"
+require_relative 'app'
+
+def main
+  app = App.new
+
+  loop do
+    display_menu
+
+    choice = gets.chomp.to_i
+
+    handle_choice(choice, app)
+
+    break if choice == 7
+  end
+
+  puts 'Goodbye!'
+end
+
+def display_menu
+  puts 'Welcome to the Rental App!'
+  puts 'Please choose an option:'
+  puts '1. List all books'
+  puts '2. List all people'
+  puts '3. Create a person'
+  puts '4. Create a book'
+  puts '5. Create a rental'
+  puts '6. List rentals for a person'
+  puts '7. Quit'
+end
+
+def handle_choice(choice, app)
+  options = {
+    1 => :list_all_books,
+    2 => :list_all_people,
+    3 => :create_person,
+    4 => :create_book,
+    5 => :create_rental,
+    6 => :list_rentals_for_person,
+    7 => :quit
+  }
+
+  if options.key?(choice)
+    send(options[choice], app)
+  else
+    puts 'Invalid choice. Please try again.'
+  end
+end
+
+def list_all_books(app)
+  app.list_all_books
+end
+
+def list_all_people(app)
+  app.list_all_people
+end
+
+def create_person(app)
+  puts "Enter the person's name:"
+  name = gets.chomp
+
+  puts "Enter the person's age:"
+  age = gets.chomp.to_i
+
+  puts "Enter the person's type (teacher or student):"
+  type = gets.chomp
```

Comment:
- [ ] According to the given video the app must ask you to input a number 1 or 2 to identify teacher or student, because currently the user can input anything which shouldn't be the case 
Yours:
![image](https://github.com/Rishi-Mishra0704/ruby-class/assets/62705947/ec74a55c-388a-4da5-bb32-810df59b67f9)
Given video:
![image](https://github.com/Rishi-Mishra0704/ruby-class/assets/62705947/30e36e28-2803-4111-a676-b66e7d9dadba)